cardinal/ecs/world.go

```diff
@@ -640,6 +666,31 @@ func (w *World) recoverGameState() (recoveredTxs *message.TxQueue, err error) {
 	return w.TickStore().Recover(w.registeredMessages)
 }
 
+func (w *World) CheckComponentSchemas() error {
+	var diffErr error
+	for _, componentMetadata := range w.registeredComponents {
+		storedSchema, err := w.redisStorage.Schema.GetSchema(componentMetadata.Name())
+		if err != nil {
+			if eris.Is(eris.Cause(err), redis.Nil) {
+				// No component schemas have been saved, skip this operation return nil error.
+				return nil
```

Comment:
So if we find that 1 component is not in redis, we can safely assume NO components are in redis? 

What about this sequence:

// Redis starts out empty.
world1 := newWorld()
world1.RegiterComponent(Alpha{})
world1.RegisterComponent(Beta{})
world1.CheckComponentSchemas() // Nothing is in redis; everything is fine.

// Redis has a record of Alpha and Beta
world2 := newWorld()
world2.RegisterComponent(Gamma{})
world2.RegisterComponent(ModifiedAlpha{})
world2.RegisterComponent(Beta{})
world2.CheckComponentSchema() // There's a problem with Alpha, but since Gamma is missing, this function will bail out early, indicating no problem.

Reply:
Yeah this is a bug. It should be continue. This function isn't used anyway. I'm deleting it.

Reply:
@jerargus The checking logic happens on Registration.